### ISSUE TITLE
Allow to preconfigure dev services container for Kubernetes

### ISSUE THREAD

**Issue by AleksanderBrzozowski** [quarkusio/quarkus#38138]
### Description

We have a Kubernetes Operator implemented with Java Operator SDK and Quarkus.

For integration tests, we leverage dev services for Kubernetes provided by Quarkus:

```yaml
  kubernetes-client:
    devservices:
      enabled: true
      flavor: k3s
      override-kubeconfig: true
```

Under the hood, a K3S Kubernetes distribution is provisioned and used for test purposes.

What do we lack from this setup is an ability to test `ValidatingWebhookConfiguration` resource from Kubernetes. We want to add a validation webhook to one of the resources that the Operator reconciles, therefore we expose a validation endpoint that is later called by Kubernetes API Server when a resource is changed.

The problem is that it is not possible to test this scenario with the current implementation, because there is no way to call the service from within Kubernetes docker container:

```
webhook: Kubernetes container --port: 8444--> Quarkus application 
```

I was able to test this scenario when I explicitly created a container and enabled `accessToHost`:

```kotlin
K3sContainer()
            .withAccessToHost(true)
            .apply { start() }
```

Then I had to expose port, like below, and a call started to work:

```kotlin
Testcontainers.exposeHostPorts(8444)
```

To sum up, I am wondering if it is possible to expose a property, for example:

```yaml
  kubernetes-client:
    devservices:
      accessToHost: true / false # new property
```

What do you think?

**Comment by geoand** [comment]
I'll leave this one to @metacosm and @csviri since they SME

**Comment by csviri** [comment]
sure, I can take a look next week
